File: 9ariniPi/src/entities/Panier.java
package entities;

import java.sql.Date;

/**
 *
 * @author Aminous
 */
public class Panier {

    private int panierID, utilisateurID;
    private Date datePanier;
    private String etatPanier;
    
    public Panier() {
    }

    public Panier(Date datePanier, String etatPanier) {
        this.datePanier = datePanier;
        this.etatPanier = etatPanier;
    }

    public Panier(int utilisateurID, Date datePanier, String etatPanier) {
        this.utilisateurID = utilisateurID;
        this.datePanier = datePanier;
        this.etatPanier = etatPanier;
    }

    public Panier(int panierID, int utilisateurID, Date datePanier, String etatPanier) {
        this.panierID = panierID;
        this.utilisateurID = utilisateurID;
        this.datePanier = datePanier;
        this.etatPanier = etatPanier;
    }

    public static Date getCurrentDate() {
        java.util.Date date = new java.util.Date();
        java.sql.Date sqlDate = new java.sql.Date(date.getTime());
        return sqlDate;
    }

    public int getPanierID() {
        return panierID;
    }

    public void setPanierID(int panierID) {
        this.panierID = panierID;
    }

    public int getUtilisateurID() {
        return utilisateurID;
    }

    public void setUtilisateurID(int utilisateurID) {
        this.utilisateurID = utilisateurID;
    }

    public Date getDatePanier() {
        return datePanier;
    }

    public void setDatePanier(Date datePanier) {
        this.datePanier = datePanier;
    }

    public String getEtatPanier() {
        return etatPanier;
    }

    public void setEtatPanier(String etatPanier) {
        this.etatPanier = etatPanier;
    }

    @Override
    public String toString() {
        return "Panier{" + "panierID=" + panierID + ", utilisateurID=" + utilisateurID + ", datePanier=" + datePanier + ", etatPanier=" + etatPanier + '}';
    }
}

File: src/Entity/Panier.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Panier
 *
 * @ORM\Table(name="panier", indexes={@ORM\Index(name="FOREIGN1", columns={"utilisateurID"})})
 * @ORM\Entity(repositoryClass="App\Repository\PanierRepository")
 */
class Panier
{
    /**
     * @var int
     *
     * @ORM\Column(name="panierID", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $panierid;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="datePanier", type="date", nullable=false)
     */
    private $datepanier;

    /**
     * @var string
     *
     * @ORM\Column(name="etatPanier", type="string", length=8, nullable=false)
     */
    private $etatpanier;

    /**
     * @var \Utilisateur
     *
     * @ORM\ManyToOne(targetEntity="Utilisateur")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="utilisateurID", referencedColumnName="id")
     * })
     */
    private $utilisateurid;

    public function getPanierid(): ?int
    {
        return $this->panierid;
    }

    public function getDatepanier(): ?\DateTimeInterface
    {
        return $this->datepanier;
    }

    public function setDatepanier(\DateTimeInterface $datepanier): self
    {
        $this->datepanier = $datepanier;

        return $this;
    }

    public function getEtatpanier(): ?string
    {
        return $this->etatpanier;
    }

    public function setEtatpanier(string $etatpanier): self
    {
        $this->etatpanier = $etatpanier;

        return $this;
    }

    public function getUtilisateurid(): ?Utilisateur
    {
        return $this->utilisateurid;
    }

    public function setUtilisateurid(?Utilisateur $utilisateurid): self
    {
        $this->utilisateurid = $utilisateurid;

        return $this;
    }


}
